hatchet/readers/caliper_native_reader.py

```diff
@@ -87,7 +87,11 @@ def _create_metric_df(self, metrics):
             if self.filename_or_caliperreader.attribute(col).is_value():
                 self.metric_cols.append(col)
         df_metrics = pd.DataFrame.from_dict(data=metrics)
-        df_new = df_metrics.groupby(df_metrics["nid"]).aggregate("first").reset_index()
+        # Aggregate on nid if timeseries data
+        if "timeseries.snapshot" in df_metrics:
+            df_new = df_metrics.groupby(df_metrics["nid"]).aggregate("first").reset_index()
```

Comment:
Is it possible that there would be multiple non-NA rows, so `aggregate("mean"` instead of taking the first value?